### PR TITLE
TimeStepper System - ComposeTimeStepperAction fix #24422

### LINKED ISSUE #24422: Throw out an error when the input time stepper(s) are not used in [TimeSteppers] block
## Bug Description

No error message is thrown out when the input time stepper(s) are not used in [TimeSteppers] block 

## Steps to Reproduce

An example as below, there is an error in the input hierarchy(A pair of '[]' is missing for the input time stepper), so the input time stepper `TimeSequenceStepper ` is not actually used. In the solve, the default time stepper is used instead. No error throws out to remind the user that the input time stepper is not used.
```
[Executioner]
  type = Transient   # Here we use the Transient Executioner
  [TimeSteppers]
     type = TimeSequenceStepper
     time_sequence  = '0 43200 86400 172800 432000 864000'
  []
  start_time = 0.0
  end_time = 864000
[]
```
## Impact

Improve user experience

### PR DESCRIPTION
**Pull request by MengnanLi91**
close #24422

### CODE REVIEW

**Comment by MengnanLi91** [comment]
@loganharbour The time stepper PR

File: framework/doc/content/syntax/Executioner/TimeSteppers/index.md [hunk 1]
```diff
@@ -19,6 +19,40 @@ This example shows the use of multiple time steppers to compose a final time ste
...
+The `[TimeSteppers]` block require one hierarchy of block for input time stepper(s).
+
+## Example of comparison between `[TimeSteppers]`and `[TimeStepper]` input
+
+Below are two examples shows the difference hierarchy between `[TimeSteppers]` and `[TimeStepper]`:
+
+`[TimeStepper]` input:
+```
+[Executioner]
+  type = Transient   # Here we use the Transient Executioner
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Executioner/TimeSteppers/index.md:32]:
```suggestion
  type = Transient
```

File: framework/doc/content/syntax/Executioner/TimeSteppers/index.md [hunk 2]
```diff
@@ -19,6 +19,40 @@ This example shows the use of multiple time steppers to compose a final time ste
...
+     time_sequence  = '0 43200 86400 172800 432000 864000'
+  []
+  start_time = 0.0
+  end_time = 864000
+[]
+```
+`[TimeSteppers]` input:
+```
+[Executioner]
+  type = Transient   # Here we use the Transient Executioner
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Executioner/TimeSteppers/index.md:44]:
```suggestion
  type = Transient
```

File: test/tests/time_steppers/time_stepper_system/tests [hunk 3]
```diff
@@ -75,7 +75,14 @@
...
       expect_err = "The following error occurred in the object \"CompositionDT\", of type \"CompositionDT\".[.\n]\sNo TimeStepper\(s\) are currently active to compute a timestep"
-      detail ='no time steppers are active to compute a timestep.'
+      detail ='no time steppers are active to compute a timestep,'
+    []
+
+    [input_hierarchy]
+      type = RunException
+      input = 'timestepper_input_error.i'
+      expect_err = "unused parameter 'Executioner/TimeSteppers/time_sequence'"
+      detail ='input time stepper(s) is not used.'
```
Comment (loganharbour) [inline test/tests/time_steppers/time_stepper_system/tests:85]:
```suggestion
      detail = 'the syntax for multiple time steppers is used without an additional nested block for the time stepper'
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — DISMISSED** [review]

**Review by GiudGiud — APPROVED** [review]

**Review by loganharbour — APPROVED** [review]

**Pull request merged**